### PR TITLE
Mostly irace related changes

### PR DESCRIPTION
**Pull request by g-skiba**

### CODE REVIEW

File: irace/test-instances-list.txt [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+# 100/100-1_100-4.tsp --time 3
+
+# berlin52
+# lust_kroA100
+# kroA150
+# kroA150
+# kroA150
+# kroA150
+# kroA150
+# kroA150
```
Comment (Starzu) [inline irace/test-instances-list.txt:23]: Add a comment about duplicates

File: src/main/scala/Main.scala [hunk 2]
```diff
@@ -270,3 +268,82 @@ object RunLoop {
...
+
+    val pheromoneConfig = PheromoneType.valueOf(conf.pheromoneType()) match
+      case PheromoneType.Basic =>
+        assert(conf.twoDimPhSize.isEmpty)
+        assert(conf.getType.isEmpty)
+        assert(conf.updateType.isEmpty)
+        PheromoneConfig(
+          conf.pheromoneType(), phDimension, phIncrement, phExtinction, phMinValue, phMaxValue,
+          takenAntsToPheromoneUpdate, new TwoDimPheromoneConfig()
+        )
```
Comment (Starzu) [inline src/main/scala/Main.scala:310]: Should we introduce subtypes of `PheromoneConfig` for various `PheromoneType`?

File: src/main/scala/project/algorithm/SingleObjectiveSolver.scala [hunk 3]
```diff
@@ -19,24 +19,25 @@ import scala.util.Random
...
+    seed: Option[Long] = None
 ) extends BaseAlgorithm {
   val solutionRepo = new SingleObjectiveSolutionRepo()
 
   override def run(resultsWriter: AcoLogger): BaseSolutionRepo = {
-    val heuristicWeights = List(1.0)
-    val pheromoneWeights = List(1.0)
+    val heuristicWeights = Array(1.0)
+    val pheromoneWeights = Array(1.0)
+    val rnd = seed.map(Random(_)).getOrElse(Random())
```
Comment (Starzu) [inline src/main/scala/project/algorithm/SingleObjectiveSolver.scala]:
```
    seed.map(Random(_)).getOrElse(Random())
```
you can make it def and reuse

File: src/main/scala/project/decision/BasicDecisionAlgorithm.scala [hunk 4]
```diff
@@ -21,27 +21,31 @@ class BasicDecisionAlgorithm[T](
...
-    @tailrec
-    def sumWeighted(acc: Double, a: Iterator[Double], b: Iterator[Double]): Double = {
-      if (a.hasNext && b.hasNext) {
-        sumWeighted(acc + a.next() * b.next(), a, b)
-      } else {
-        acc
+    def sumWeighted(a: Array[Double], b: Array[Double]): Double = {
+      @tailrec
+      def sumWeightedInd(acc: Double, ind1: Int, ind2: Int): Double = {
+        if (ind1 < a.length && ind2 < b.length) {
```
Comment (Starzu) [inline src/main/scala/project/decision/BasicDecisionAlgorithm.scala]: Do we ever expect these sizes to be different?

**Review by Starzu — APPROVED** [review]

File: src/main/scala/Main.scala (hunk 2 above)
Comment (g-skiba) [reply src/main/scala/Main.scala:310]: That would be a good idea probably but the yaml configuration is problematic. I'll leave it as is for now.

**Pull request merged**